server/src/test/java/org/opensearch/extensions/rest/RestExecuteOnExtensionTests.java

```diff
@@ -25,19 +28,38 @@ public class RestExecuteOnExtensionTests extends OpenSearchTestCase {
     public void testRestExecuteOnExtensionRequest() throws Exception {
         Method expectedMethod = Method.GET;
         String expectedUri = "/test/uri";
-        RestExecuteOnExtensionRequest request = new RestExecuteOnExtensionRequest(expectedMethod, expectedUri);
+        PrincipalIdentifierToken expectedRequestIssuerIdentity = new PrincipalIdentifierToken("requestor-identifier");
```

Comment:
This doesn't conform to the format expected by `extractToken()`.  (Which indicates that method is not tested...)

Reply:
This identifier stores the token object (a string), and its wire-safe since it doesn't include any of identity information (principal/subject).

IMO this is correct for this file; however for `RestSendToExtensionAction.java` something should have broken.

Reply:
A separate check should be implemented to assert the validity of the token. However, I'm unsure about whether it should be implemented now or once encryption/decryption is in place

Reply:
I have implemented a simple check that checks validity of token based on current input format

Reply:
I have changed the way the PrincipalIdentifierToken is instantiated in this class as we don't expected it to be instantiated with a raw string ever. Instead it should always call extensionTokenProcessor.generateToken(principal) to return an object.